### PR TITLE
ROLLUP-203/deployment-script

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Create deployment script
- Modify current scripts to use docker containers instead of local environment

## Why

- Facilitate the deployment of the environments

## Refs

- [ROLLUP-207](https://rsklabs.atlassian.net/browse/ROLLUP-207)
- [ROLLUP-203](https://rsklabs.atlassian.net/browse/ROLLUP-203)

### CODE REVIEW

File: infrastructure/zk/src/server.ts [hunk 1]
```diff
@@ -3,41 +3,61 @@ import * as utils from './utils';
 import * as env from './env';
 import fs from 'fs';
 import * as db from './db/db';
+import * as deploy from './deploy';
 
 import { ethers } from 'ethers';
 
-export async function core() {
+export async function core(docker: boolean) {
```
Comment (antomor) [inline infrastructure/zk/src/server.ts:10]: I'd specify false as  default value, so we don't need to change all the previous usages

File: infrastructure/zk/src/server.ts [hunk 2]
```diff
@@ -3,41 +3,61 @@ import * as utils from './utils';
...
+    } else {
+        await utils.spawn(
+            'cargo run --bin zksync_server --release -- --components=eth-sender,witness-generator,forced-exit,prometheus,core,rejected-task-cleaner,fetchers,prometheus-periodic-metrics'
+        );
+    }
 }
 
-export async function web3Node() {
-    await utils.spawn('cargo run --bin zksync_server --release -- --components=web3-api');
+export async function web3Node(docker: boolean) {
```
Comment (antomor) [inline infrastructure/zk/src/server.ts:22]: I'd specify false as default value, so we don't need to change all the previous usages

File: infrastructure/zk/src/server.ts [hunk 3]
```diff
@@ -3,41 +3,61 @@ import * as utils from './utils';
...
+    } else {
+        await utils.spawn('cargo run --bin zksync_server --release -- --components=web3-api');
+    }
 }
 
-export async function apiNode() {
-    await utils.spawn(
-        'cargo run --bin zksync_server --release -- --components=web3-api,rest-api,rpc-api,rpc-websocket-api'
-    );
+export async function apiNode(docker: boolean) {
```
Comment (antomor) [inline infrastructure/zk/src/server.ts:30]: I'd specify false as  default value, so we don't need to change all the previous usages

File: infrastructure/zk/src/server.ts [hunk 4]
```diff
@@ -3,41 +3,61 @@ import * as utils from './utils';
...
+        await deploy.dockerUp('server-api');
+    } else {
+        await utils.spawn(
+            'cargo run --bin zksync_server --release -- --components=web3-api,rest-api,rpc-api,rpc-websocket-api'
+        );
+    }
 }
 
-export async function server() {
+export async function server(docker: boolean) {
```
Comment (antomor) [inline infrastructure/zk/src/server.ts:40]: I'd specify false as  default value, so we don't need to change all the previous usages

File: infrastructure/zk/src/server.ts [hunk 5]
```diff
@@ -3,41 +3,61 @@ import * as utils from './utils';
...
-    await utils.spawn('cargo run --bin zksync_server --release');
+    if (docker) {
+        await deploy.dockerUp('server');
+    } else {
+        await utils.spawn('cargo run --bin zksync_server --release');
+    }
 }
 
-export async function genesis() {
+export async function genesis(docker: boolean) {
```
Comment (antomor) [inline infrastructure/zk/src/server.ts:53]: I'd specify false as  default value, so we don't need to change all the previous usages

File: infrastructure/zk/src/run/run.ts [hunk 6]
```diff
@@ -94,10 +93,13 @@ export async function tokenInfo(address: string) {
 }
 
 // installs all dependencies and builds our js packages
-export async function yarn() {
+export async function yarn(sdk: boolean) {
```
Comment (antomor) [inline infrastructure/zk/src/run/run.ts:96]: I'd specify true as default value, so we don't need to change all the previous usages

File: infrastructure/zk/src/run/run.ts [hunk 7]
```diff
@@ -49,8 +49,7 @@ export async function deployERC20(command: 'dev' | 'new', name?: string, symbol?
         }
         if (feeTicker) {
             feeTicker.fee_ticker.unconditionally_valid_tokens = [
-                ...unconditionallyValidTokens,
-                ...deployedUnconditionallyValidAddresses
+                ...new Set([...unconditionallyValidTokens, ...deployedUnconditionallyValidAddresses])
```
Comment (antomor) [inline infrastructure/zk/src/run/run.ts:52]: 👍🏼

File: infrastructure/zk/src/init.ts
Comment (antomor) [inline infrastructure/zk/src/init.ts:1]: Should we move the functions used elsewhere in utils.ts file?

**Review by antomor — CHANGES_REQUESTED** [review]
@franciscotobar I haven't tested everything locally, but I've left some comments there. Thank you for the changes.

File: docker-compose.deploy.yml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    ports:
+      - 3001:3001
+      - 3002:3002
+      - 3030:3030
+      - 3031:3031
+      - 3032:3032
+      - 3312:3312
+      - 8080:8080
+      - 8088:8088
+      - 8090:8090
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docker-compose.deploy.yml:15]: updated

File: docs/deployment.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
+# Deployment guide
+
+This document covers deployment-related actions in rif-rollup. The deployment script provides multiple utilities to
+prepare and deploy the different rif-rollup components.
+
+The deployment script can take advantage of docker and reduce the amount of components to be build. The deployment
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+script does not take into account the RSKj node and the database.
+
+## Prerequisites
+
+By default the rif-rollup system will use the `dev` environments, to use different environments we need to compile the
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+configuration folder with the command:
+
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md:12]: updated

File: docs/deployment.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+```
+zk config compile testnet
+```
+
+**Note:** This command looks for the folder with the same name inside the `./etc/env/`.
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+Once the environment is compile, we can switch to it by executing the command:
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+```
+zk env testnet
+```
+
+## Environment
+
+The deployment script needs the `DEPLOYMENT=true` to perform the actions properly. By activating this flag in the
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: removed

File: docs/deployment.md [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+environments from the system, any `env` profile will be overriden by `./etc/env/deploy.env` file.
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: removed

File: docs/deployment.md [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+
+To prepare the environment we just need to execute the command:
+
+```
+DEPLOYMENT=true zk deploy prepare-environment
+```
+
+### Docker compose
+
+The deployment script use the `./docker-compose.deploy.yml` to execute the different commands, due to `docker-compose`
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+limitations; each time that we change our environment, the `env_file` section needs to be updated manually in each
+service of the file.
+
+## Server
+
+To deploy the components that are needed by the `server` we need to prepare the environment, to prepare the environment
+we just need to execute the command:
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md:43]: updated

File: docs/deployment.md [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+Once the environment is ready, the `server` can be deployed using the following command:
+
+```
+DEPLOYMENT=true zk server
+```
+
+## Prover
+
+To deploy the components that are needed by the `prover/dummy-prover` we need to prepare the environment, to prepare the
+environment we just need to execute the command:
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+## Server
+
+To deploy the components that are needed by the `server` we need to prepare the environment, to prepare the environment
+we just need to execute the command:
+
+```
+zk deploy --docker prepare-server
+```
+
+Once the environment is ready, the `server` can be deployed using the following command:
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md:49]: updated

File: docs/deployment.md [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+
+```
+zk server --docker
+```
+
+## Prover
+
+To deploy the components that are needed by the `dummy-prover` we need to prepare the environment, to prepare the
+environment we just need to execute the command:
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md:58]: updated

File: docs/deployment.md [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+
+```
+zk deploy --docker prepare-prover
+```
+
+To use the `dummy-prover` we need to enable it, prior enabling it, the contracts must be already deployed. During the
+deployment of the contracts, a `genesis.log` file is created with the `CONTRACTS_GENESIS_ROOT` inside of it. This file
+needed to execute the command:
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md:66]: updated

File: docs/deployment.md [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+
+```
+zk dummy-prover --docker enable
+```
+
+Once the `dummy-prover` is enabled, it can be deployed executing the following command:
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md:72]: updated

File: infrastructure/zk/src/deploy.ts [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+    const DIRECTORY = './etc/env';
+
+    const extra = process.env.OVERRIDE ? `EXTRA_ENV_FILE=${DIRECTORY}/${process.env.OVERRIDE}.env` : '';
+
+    return `ENV_FILE=${DIRECTORY}/${process.env.ZKSYNC_ENV}.env ${extra}`;
+}
+
+export const command = new Command('deploy')
+    .option('--docker', 'use docker container instead of local environment')
+    .option('--sdk', 'include sdk packages')
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply infrastructure/zk/src/deploy.ts:58]: Not really, actually we just need to build it for testing purposes, I can always setup to false.

File: infrastructure/zk/src/init.ts
Comment (franciscotobar) [reply infrastructure/zk/src/init.ts:1]: what do you mean with this?

File: etc/env/deploy.env [hunk 24]
```diff
@@ -0,0 +1,10 @@
+ETH_CLIENT_WEB3_URL=http://host.docker.internal:4444
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply etc/env/deploy.env:1]: updated

File: etc/env/base/forced_exit_requests.toml [hunk 25]
```diff
@@ -47,3 +47,5 @@ blocks_check_amount=10
 # How often we want to poll the RSK node (in milliseconds).
 eth_node_poll_interval=300
 
+# Addresses of the RSK node API, separated by comma. Consider change the web3_url to a particular IP pointing to an RSK full node.
+web3_url="http://127.0.0.1:4444"
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply etc/env/base/forced_exit_requests.toml:51]: Ill be removing it and moving to a different place. But we may need to discuss it :)

File: docker/server/Dockerfile [hunk 26]
```diff
@@ -1,16 +1,21 @@
...
-EXPOSE 3030
+EXPOSE 3001
 EXPOSE 3002
+EXPOSE 3030
+EXPOSE 3031
+EXPOSE 3032
+EXPOSE 3312
+EXPOSE 8080
+EXPOSE 8088
+EXPOSE 8090
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docker/server/Dockerfile:18]: Writing EXPOSE in your Dockerfile, is merely a hint that a certain port is useful. Docker won’t do anything with that information by itself. So I can remove if you think that is not necessary, but I think that may help to understand it easy.

File: infrastructure/zk/src/run/run.ts (hunk 6 above)
Comment (franciscotobar) [reply infrastructure/zk/src/run/run.ts:96]: updated

File: infrastructure/zk/src/server.ts (hunk 5 above)
Comment (franciscotobar) [reply infrastructure/zk/src/server.ts:53]: updated

File: infrastructure/zk/src/server.ts (hunk 4 above)
Comment (franciscotobar) [reply infrastructure/zk/src/server.ts:40]: updated

File: infrastructure/zk/src/server.ts (hunk 3 above)
Comment (franciscotobar) [reply infrastructure/zk/src/server.ts:30]: updated

File: infrastructure/zk/src/server.ts (hunk 2 above)
Comment (franciscotobar) [reply infrastructure/zk/src/server.ts:22]: updated

File: docker/server/Dockerfile (hunk 26 above)
[the remark this answers is not in the document]
Comment (antomor) [reply docker/server/Dockerfile:18]: My only concern is that we need to add a comment to clarify that we need to change them in 3 places if necessary

File: docker-compose.deploy.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (antomor) [reply docker-compose.deploy.yml]: could we also clarify with a comment that if we change them here, we need to update the other places?

File: docs/deployment.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (antomor) [reply docs/deployment.md]: I think this is still valid

File: docs/deployment.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (antomor) [reply docs/deployment.md]: it looks like it's still vallid

File: docker-compose.deploy.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docker-compose.deploy.yml]: Im not sure about the comment, Ill want to discuss it. If someone is going to change the docker-compose file, that person should know what is doing. I included a comment referring to what API refers to.

File: docker/server/Dockerfile (hunk 26 above)
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docker/server/Dockerfile:18]: Ill be removing it from here, since its just a guide for the one that see the Dockerfile, but this is dynamic based on the Environment variables.

File: docs/deployment.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: you are right, updated.

File: docs/deployment.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md [hunk 27]
```diff
@@ -0,0 +1,111 @@
...
+As part of the deployment we need to deploy the `dev-ticker` and `dev-liquidity-token-watcher` by executing the
+following command:
+
+```
+docker-compose -f docker-compose.deploy.yml up -d dev-ticker dev-liquidity-token-watcher
+```
+
+## Example
+
+To deploy a local environment we need to execute the commands in the following order:
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md [hunk 28]
```diff
@@ -0,0 +1,159 @@
...
+```
+
+This command looks for the folder with the same name (`<env_name>`) inside the `./etc/env/`. Once the environment is
+compiled, we can select it with the command:
+
+```
+zk env <env_name>
+```
+
+### Override
```
Comment (jurajpiar) [inline docs/deployment.md:24]: I'd call it CUSTOM_ENV, or ENV_OVERRIDE, but override is tad ambiguous.

**Review by jurajpiar — APPROVED** [review]
The only issue I have is with the deploying erc20 in the deploy process. If this is a command that will be used in production then we should not deploy the dev tokens right

File: docker-compose.deploy.yml [hunk 29]
```diff
@@ -0,0 +1,103 @@
...
+      - "9975:9975"
+    volumes:
+      - ./etc/tokens/:/etc/tokens
+
+  dev-ticker:
+    image: "matterlabs/dev-ticker:latest"
+    ports:
+      - "9876:9876"
+    volumes:
+      - ./etc/tokens/:/etc/tokens
```
Comment (jurajpiar) [inline docker-compose.deploy.yml:71]: do we need these in production?

File: infrastructure/zk/src/deploy.ts [hunk 30]
```diff
@@ -0,0 +1,73 @@
...
+    await announced('Checking PLONK setup', run.plonkSetup());
+    await announced('Unpacking verification  keys', run.verifyKeys.unpack());
+}
+
+async function prepareServer(docker: boolean) {
+    await prepareEnvironment();
+    await prepareKeys();
+    await announced('Setting up database', db.setup());
+    await announced('Building contracts', contract.build());
+    await announced('Deploying localhost ERC20 tokens', run.deployERC20('dev'));
```
[the remark this answers is not in the document]
Comment (jurajpiar) [reply infrastructure/zk/src/deploy.ts:25]: not a production

File: docker-compose.deploy.yml (hunk 29 above)
Comment (jurajpiar) [reply docker-compose.deploy.yml:71]: not for prod

File: infrastructure/zk/src/init.ts
Comment (antomor) [reply infrastructure/zk/src/init.ts:1]: Sorry for being unclear, I mean that the functions exposed in the `init.ts` could be moved to `utils.ts` since they're also used else where
(NIT)

File: docker-compose.yml [hunk 31]
```diff
@@ -25,9 +25,7 @@ services:
       - ./docker/rskj/data:/var/lib/rsk/.rsk
   
   dev-ticker:
-    build:
-      context: .
-      dockerfile: ./docker/dev-ticker/Dockerfile
+    image: "rsksmart/rollup-dev-ticker:0.0.1-beta"
```
Comment (antomor) [inline docker-compose.yml:28]: I guess this is temporary, isn't it?

File: docs/development.md [hunk 32]
```diff
@@ -25,11 +25,11 @@ zk init
 This command will do the following:
 
 - Generate `$ZKSYNC_HOME/etc/env/dev.env` file with settings for the applications.
-- Initialize docker containers with `geth` Rootstock node and `postgres` database for local development.
+- Initialize docker containers with `RSKj` node and `postgres` database for local development.
 - Download and unpack files for cryptographical backend (`circuit`).
 - Generate required smart contracts.
 - Compile all the smart contracts.
-- Deploy smart contracts to the local Rootstock network.
+- Deploy smart contracts to the local RSKj network.
```
[the remark this answers is not in the document]
Comment (antomor) [reply docs/development.md:32]: 👆🏽

File: docker-compose.deploy.yml [hunk 33]
```diff
@@ -0,0 +1,71 @@
...
+        target: /keys/
+
+  dev-liquidity-token-watcher:
+    image: "matterlabs/dev-liquidity-token-watcher:latest"
+    env_file:
+      - ./etc/env/deploy.env
+    ports:
+      - "9975:9975"
+    volumes:
+      - ./etc/tokens/:/etc/tokens
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docker-compose.deploy.yml]: updated

File: infrastructure/zk/src/init.ts
Comment (franciscotobar) [reply infrastructure/zk/src/init.ts:1]: updated

File: docs/deployment.md [hunk 34]
```diff
@@ -0,0 +1,159 @@
...
+
+To simplify the way that some of the environment variables can be updated, the override functionality was introduced.
+
+This functioanlity consist on setting the `OVERRIDE=<file_name>` environment variable, that will look for the
```
[the remark this answers is not in the document]
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docs/deployment.md (hunk 28 above)
Comment (franciscotobar) [reply docs/deployment.md]: updated

File: docker-compose.yml (hunk 31 above)
Comment (franciscotobar) [reply docker-compose.yml]: updated

File: etc/env/base/forced_exit_requests.toml (hunk 25 above)
[the remark this answers is not in the document]
Comment (franciscotobar) [reply etc/env/base/forced_exit_requests.toml]: updated

File: core/bin/zksync_api/src/bin/providers/dev_liquidity_provider.rs [hunk 35]
```diff
@@ -237,7 +237,9 @@ mod handlers_tests {
     #[actix_web::test]
     async fn get_coin_contract() {
         let expected_volume = BigDecimal::from_f32(2.34567).unwrap();
-        let token_address = "0xe978d5775bdCfB872c3Bf919Ce4169cDA6f8E271".to_string(); // an arbitrary, pre-generated address
+        let token_address = "0xe978d5775bdCfB872c3Bf919Ce4169cDA6f8E271"
+            .to_string()
+            .to_ascii_lowercase(); // an arbitrary, pre-generated address
```
Comment (franciscotobar) [inline core/bin/zksync_api/src/bin/providers/dev_liquidity_provider.rs:242]: @jurajpiar I had to change this to lowercase, since in the implementation is done like that.

File: etc/env/deploy.env [hunk 36]
```diff
@@ -0,0 +1,16 @@
+# In Mac / Windows replace the 172.17.0.1 with host.docker.internal
+DATABASE_URL=postgres://postgres@172.17.0.1/plasma
+FORCED_EXIT_REQUESTS_WEB3_URL=http://172.17.0.1:4444
```
Comment (david-iov) [inline etc/env/deploy.env:3]:
```suggestion
DATABASE_URL=postgres://$DB_USER:$DB_PASSWORD@$DB_HOST:5432/$DB_NAME
FORCED_EXIT_REQUESTS_WEB3_URL=http://localhost:4444
```
Can we have this instead? I'll have a script in the server that updates this file with the variables.

File: etc/env/base/forced_exit_requests.toml [hunk 37]
```diff
@@ -46,4 +46,3 @@ blocks_check_amount=10
 
 # How often we want to poll the RSK node (in milliseconds).
 eth_node_poll_interval=300
-
```
Comment (antomor) [inline etc/env/base/forced_exit_requests.toml:49]: @franciscotobar Could we add an empty line here as it was before?

File: etc/env/deploy.env (hunk 36 above)
Comment (franciscotobar) [reply etc/env/deploy.env:3]: you can create your own file and just update the `ENV_OVERRIDE` variable.

Comment (david-iov) [reply etc/env/deploy.env:3]: Also, the localhost in `FORCED_EXIT_REQUESTS_WEB3_URL` can be variable too.

File: core/bin/zksync_api/src/bin/providers/dev_liquidity_provider.rs (hunk 35 above)
Comment (jurajpiar) [reply core/bin/zksync_api/src/bin/providers/dev_liquidity_provider.rs:242]: 👍

**Review by jurajpiar — DISMISSED** [review]
LG

File: core/lib/crypto/src/merkle_tree/parallel_smt.rs [hunk 38]
```diff
@@ -882,7 +882,7 @@ mod tests {
         for &(idx, value) in &test_vector[..] {
             let merkle_proof = tree.merkle_path(idx);
 
-            let hasher = TestHasher::default();
+            let hasher = TestHasher;
```
Comment (antomor) [inline core/lib/crypto/src/merkle_tree/parallel_smt.rs:885]: @franciscotobar why this change was necessary for this PR?

File: docs/deployment.md [hunk 39]
```diff
@@ -0,0 +1,228 @@
...
+zk server
+```
+
+### Forced Exit
+
+During the run process of the server, the preparation for the `forced exit requests account` is done. Depending where
+the `zk server` command is going to be executed, the `ETH_CLIENT_WEB3_URL` could be different, therefore an additional
+environment variable was created to provide more flexibility `FORCED_EXIT_REQUESTS_WEB3_URL`.
+
+To validate that the preparation for the `forced exit requests account` can reach the web3 node, we can execute the
```
Comment (antomor) [inline docs/deployment.md:78]: 👏🏽

File: etc/env/deploy.env [hunk 40]
```diff
@@ -0,0 +1,19 @@
+# In Mac / Windows replace the host.docker.internal with host.docker.internal
```
Comment (antomor) [inline etc/env/deploy.env:1]: I guess it is
```suggestion
# In Mac / Windows replace 172.17.0.1 with host.docker.internal
```

File: etc/env/deploy.env
Comment (antomor) [inline etc/env/deploy.env:1]: shouldn't we replace all the `host.docker.internal` with `172.17.0.1`?

File: etc/env/mainnet/forced_exit_requests.toml
Comment (antomor) [inline etc/env/mainnet/forced_exit_requests.toml:1]: Again, here I see as we removed the last line 🤔

File: etc/env/qa/private.toml
Comment (antomor) [inline etc/env/qa/private.toml:1]: I think we should remove this file from here, so we could copy the right one when necessary.

File: etc/test_config/sdk/test-vectors.json
Comment (antomor) [inline etc/test_config/sdk/test-vectors.json:1]: what's the purpose of this file, and why did we change it?

File: infrastructure/zk/src/docker.ts [hunk 41]
```diff
@@ -43,9 +44,7 @@ async function _build(image: string, tag: string) {
     }
 
     const imageTag = `-t rsksmart/rollup-${image}:${tag}`;
-    await utils.spawn(
-        `DOCKER_BUILDKIT=1 docker build ${imageTag} --platform linux/amd64 -f ./docker/${image}/Dockerfile .`
-    );
+    await utils.spawn(`DOCKER_BUILDKIT=1 docker build ${imageTag} -f ./docker/${image}/Dockerfile .`);
```
Comment (antomor) [inline infrastructure/zk/src/docker.ts:47]: Why did we remove the platform argument?

File: docker/environment/Dockerfile [hunk 42]
```diff
@@ -7,7 +7,7 @@ RUN apt-get update; apt-get install -y make bash git gnupg openssl libssl-dev gc
 
 #install docker engine
 RUN curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -
-RUN add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/debian $(lsb_release -cs) stable"
+RUN add-apt-repository "deb https://download.docker.com/linux/debian $(lsb_release -cs) stable"
```
Comment (antomor) [inline docker/environment/Dockerfile:10]: why are we removing the architecture?

File: infrastructure/zk/src/docker.ts [hunk 43]
```diff
@@ -67,19 +67,40 @@ export async function push(image: string, tag: string) {
...
 
+export async function deployUp(service: string) {
+    console.log('WARNING! Using docker!');
+    await utils.spawn(`docker-compose -f docker-compose.deploy.yml up ${service}`);
+}
+
+export async function deployRun(command: string) {
+    console.log('WARNING! Using docker!');
+    await utils.spawn(`docker-compose -f docker-compose.deploy.yml run ${command}`);
+}
```
Comment (antomor) [inline infrastructure/zk/src/docker.ts:87]: NIT (sorry) 😅: we could define `console.log('WARNING! Using docker!');` just once.

File: infrastructure/zk/src/server.ts
Comment (antomor) [inline infrastructure/zk/src/server.ts:1]: NIT: avoid the else statements of all the `withDocker` functions by including a return statement

File: etc/test_config/sdk/test-vectors.json
Comment (antomor) [reply etc/test_config/sdk/test-vectors.json:1]: I think the answer is [here](https://github.com/rsksmart/rif-rollup/pull/87/files#r1274747821)

File: etc/env/qa/private.toml
Comment (franciscotobar) [reply etc/env/qa/private.toml:1]: I dont understand this comment. I already updated this file.

File: etc/test_config/sdk/test-vectors.json
Comment (franciscotobar) [reply etc/test_config/sdk/test-vectors.json:1]: to fix the unit test

File: infrastructure/zk/src/docker.ts (hunk 41 above)
Comment (franciscotobar) [reply infrastructure/zk/src/docker.ts:47]: to avoid having issues while building the docker image in Mac.

File: infrastructure/zk/src/docker.ts (hunk 43 above)
Comment (franciscotobar) [reply infrastructure/zk/src/docker.ts:87]: updated

File: infrastructure/zk/src/server.ts
Comment (franciscotobar) [reply infrastructure/zk/src/server.ts:1]: updated

File: etc/env/deploy.env
Comment (franciscotobar) [reply etc/env/deploy.env:1]: updated

File: etc/env/deploy.env (hunk 40 above)
Comment (franciscotobar) [reply etc/env/deploy.env:1]: updated

File: core/lib/crypto/src/merkle_tree/parallel_smt.rs (hunk 38 above)
Comment (franciscotobar) [reply core/lib/crypto/src/merkle_tree/parallel_smt.rs:885]: to fix the lint process

File: etc/env/deploy.env (hunk 36 above)
Comment (franciscotobar) [reply etc/env/deploy.env:3]: We already discussed and address it

File: init.sh [hunk 44]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+DEPLOY_ENVIRONMENT="${DEPLOY_ENVIRONMENT:=dev}" 
+
```
Comment (david-iov) [inline init.sh:4]:
```suggestion
export ZKSYNC_HOME=$PWD
export PATH=$ZKSYNC_HOME/bin:$PATH
```
Please add the zk installation here

File: docs/deployment.md [hunk 45]
```diff
@@ -0,0 +1,228 @@
...
+1. Create the `dev` environment.
+
+```
+zk config compile dev
+```
+
+2. Initialize docker containers with `rollup` environent, `RSKj` node`postgres` database for local development.
+
+```
+docker-compose -f docker-compose.deploy.yml up -d rskj postgres rollup
```
Comment (antomor) [inline docs/deployment.md:191]: @franciscotobar  This command raise an error because in `docker-compose.deploy.yml`, we don't expose any port for the postgres service

**Review by david-iov — APPROVED** [review]
LGTM

File: docker-compose.deploy.yml
[the remark this answers is not in the document]
Comment (david-iov) [reply docker-compose.deploy.yml:1]: Yeah, that's okay to include.

File: docs/deployment.md (hunk 45 above)
Comment (franciscotobar) [reply docs/deployment.md:191]: updated

**Review by antomor — APPROVED** [review]
LGTM, thanks @franciscotobar and @david-iov for your work 🙏🏼

File: docker-compose.deploy.yml
[the remark this answers is not in the document]
Comment (antomor) [reply docker-compose.deploy.yml:1]: we decided not to apply those changes for now
